lib/pbench/server/api/resources/users_api.py

```diff
@@ -8,12 +9,93 @@
 import jwt
 from sqlalchemy.exc import IntegrityError, SQLAlchemyError
 
+from pbench.server import PbenchServerConfig
 from pbench.server.api.auth import Auth
 from pbench.server.database.models.active_tokens import ActiveTokens
 from pbench.server.database.models.server_config import ServerConfig
 from pbench.server.database.models.users import User
 
 
+class AuthenticationSchema(Resource):
+    """
+    Pbench API to return authentication parameters required for the client to
+    authenticate the user with OIDC broker.
+    """
+
+    AVAILABALE_SCHEMA = ["prod", "dev", "test"]
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        self.server_config = config
+        self.logger = logger
+
+    def get(self):
+        """
+        Returns the authentication schema parameters.
+
+        Client can request a specific authorization schema paramters required
+        to authenticate an end-user. However, if the schema is not provided we
+        return the production schema to carry out the user authentication.
+
+        Currently we only support the authentication schemas named mentioned in
+        AVAILABLE_SCHEMA.
+
+        Schema name is required to be a URI parameter. E.g.
+        GET pbench_server_rest_uri/authentication?schema=[schema_name]"
+        :return: JSON Payload
+            Success: 200,
+            response_object = {
+                "schema": [schema-name],
+                "auth_realm": [realm-name],
+                "auth_client": [client-name],
+                "auth_issuer": [auth-issuer],
+                "auth_secret": [secret-string]
+                # empty auth secret means the auth client does not
+                # need a secret to communicate with an auth issuer
+            }
+            Failure: <status_Code>,
+            response_object = {
+                "message": "failure message"
+            }
+        """
+        disabled = ServerConfig.get_disabled()
+        if disabled:
+            abort(HTTPStatus.SERVICE_UNAVAILABLE, **disabled)
```

Comment:
We need to be able to authenticate to a server even when the service is disabled. The login/logout APIs are excluded from this test (see earlier comment about using `ApiBase` infrastructure!), and to authenticate the client needs to know *how* to authenticate, so this API needs to be excluded from the test.